Tests/OCTVideoEngineTests.m

```diff
@@ -145,4 +151,81 @@ - (void)testReceiveVideoFrame
     });
 }
 
+- (void)_performYUVCheckWithPlanes:(uint8_t * [3])planes strides:(OCTToxAVStrideData[3])strides
+{
+    OCTToxAVVideoWidth width = 30;
+    OCTToxAVVideoHeight height = 30;
+    self.videoEngine.videoView = OCMClassMock([OCTVideoView class]);
+    id ciMock = OCMClassMock([CIImage class]);
+
+    __block BOOL pass = NO;
+
+    OCMStub([ciMock imageWithCVPixelBuffer:[OCMArg anyPointer]]).andDo(^(NSInvocation *invocation) {
+        CVPixelBufferRef pb = NULL;
+        [invocation getArgument:&pb atIndex:2];
+
+        XCTAssertNotNil((__bridge id)pb);
+        XCTAssertEqual(CVPixelBufferGetPixelFormatType(pb), kCVPixelFormatType_420YpCbCr8BiPlanarVideoRange);
+
+        // now we compare the buffer contents with the known good result
+        CVPixelBufferLockBaseAddress(pb, kCVPixelBufferLock_ReadOnly);
+        uint8_t *y = CVPixelBufferGetBaseAddressOfPlane(pb, 0);
+        uint8_t *uv = CVPixelBufferGetBaseAddressOfPlane(pb, 1);
+
+        XCTAssertEqual(memcmp(y, test_good_y, sizeof(test_good_y)), 0);
+        XCTAssertEqual(memcmp(uv, test_good_uv, sizeof(test_good_uv)), 0);
+
+        pass = YES;
+
+        void *ret = nil;
+        [invocation setReturnValue:&ret];
+    });
+
+    [self.videoEngine receiveVideoFrameWithWidth:width
+                                          height:height
+                                          yPlane:planes[0]
+                                          uPlane:planes[1]
+                                          vPlane:planes[2]
+                                         yStride:strides[0]
+                                         uStride:strides[1]
+                                         vStride:strides[2]
+                                    friendNumber:10];
+
+    dispatch_sync(self.videoEngine.processingQueue, ^{
+        XCTAssertTrue(pass);
+    });
+}
+
+- (void)testReceivePackedVideoFrame
+{
+    uint8_t *planes[3] = {test_str83p_y, test_str83p_u, test_str83p_v};
+    OCTToxAVStrideData strides[3] = {30, 15, 15};
+
+    [self _performYUVCheckWithPlanes:planes strides:strides];
```

Comment:
The CIImage code works just fine on OS X so I've removed the guard around that instead of putting new ones on the tests.